File: Algorithms/Graph/QuickFind.py
class UnionFind():

    def __init__(self, size):
        self.rootArray = [i for i in range(size)]

    def find(self, v):
        return self.rootArray[v]
    
    def union(self, x, y):
        rootX = self.find(x) # as head
        rootY = self.find(y)
        if rootX != rootY:
            for i in range(len(self.rootArray)):
                if self.rootArray[i] == rootY:
                    self.rootArray[i] = rootX
    
    def connected(self, x,  y):
        return self.find(x) == self.find(y) 

# Test Case
uf = UnionFind(10)
# 1-2-5-6-7 3-8-9 4
uf.union(1, 2)
uf.union(2, 5)
uf.union(5, 6)
uf.union(6, 7)
uf.union(3, 8)
uf.union(8, 9)
print(uf.connected(1, 5))  # true
print(uf.connected(5, 7))  # true
print(uf.connected(4, 9))  # false
# 1-2-5-6-7 3-8-9-4
uf.union(9, 4)
print(uf.connected(4, 9))  # true

File: Algorithms/Graph/QuickFind.java
package Algorithms.Graph;

class UnionFind {
    private int[] rootArray;

    public UnionFind(int size) {
        rootArray = new int[size];
        for (int i = 0; i < size; i++) {
            rootArray[i] = i;
        }
    }

    public int find(int v) {
        return rootArray[v];
    }

    public void union(int x, int y) {
        int rootX = find(x);
        int rootY = find(y);
        if (rootX != rootY) {
            for (int i = 0; i < rootArray.length; i++) {
                if (rootArray[i] == rootY) {
                    rootArray[i] = rootX;
                }
            }
        }
    }
    
    public boolean connected(int x, int y) {
        return find(x) == find(y);
    }
}

public class QuickFind {
    public static void main(String[] args) throws Exception {
        UnionFind uf = new UnionFind(10);
        // 1-2-5-6-7 3-8-9 4
        uf.union(1, 2);
        uf.union(2, 5);
        uf.union(5, 6);
        uf.union(6, 7);
        uf.union(3, 8);
        uf.union(8, 9);
        System.out.println(uf.connected(1, 5)); // true
        System.out.println(uf.connected(5, 7)); // true
        System.out.println(uf.connected(4, 9)); // false
        // 1-2-5-6-7 3-8-9-4
        uf.union(9, 4);
        System.out.println(uf.connected(4, 9)); // true
    }
}